nemo/collections/asr/parts/utils/asr_confidence_utils.py

```diff
@@ -18,46 +18,164 @@
 
 
+class ConfidenceMeasureConstants:
+    NAMES = ("max_prob", "entropy")
+    ENTROPY_TYPES = ("gibbs", "tsallis", "renyi")
+    ENTROPY_NORMS = ("lin", "exp")
+
+    @classmethod
+    def print(cls):
+        return (
+            cls.__name__
+            + ": "
+            + str({"NAMES": cls.NAMES, "ENTROPY_TYPES": cls.ENTROPY_TYPES, "ENTROPY_NORMS": cls.ENTROPY_NORMS})
+        )
+
+
+class ConfidenceConstants:
+    AGGREGATIONS = ("mean", "min", "max", "prod")
+
+    @classmethod
+    def print(cls):
+        return cls.__name__ + ": " + str({"AGGREGATIONS": cls.AGGREGATIONS})
+
+
 @dataclass
-class ConfidenceMethodConfig:
+class ConfidenceMeasureConfig:
+    """A Config which contains the measure name and settings to compute per-frame confidence scores.
+
+    Args:
+        name: The measure name (str).
+            Supported values:
+                - 'max_prob' for using the maximum token probability as a confidence.
+                - 'entropy' for using a normalized entropy of a log-likelihood vector.
+
+        entropy_type: Which type of entropy to use (str).
+            Used if confidence_measure_cfg.name is set to `entropy`.
+            Supported values:
+                - 'gibbs' for the (standard) Gibbs entropy. If the alpha (α) is provided,
+                    the formula is the following: H_α = -sum_i((p^α_i)*log(p^α_i)).
+                    Note that for this entropy, the alpha should comply the following inequality:
+                    (log(V)+2-sqrt(log^2(V)+4))/(2*log(V)) <= α <= (1+log(V-1))/log(V-1)
+                    where V is the model vocabulary size.
+                - 'tsallis' for the Tsallis entropy with the Boltzmann constant one.
+                    Tsallis entropy formula is the following: H_α = 1/(α-1)*(1-sum_i(p^α_i)),
+                    where α is a parameter. When α == 1, it works like the Gibbs entropy.
+                    More: https://en.wikipedia.org/wiki/Tsallis_entropy
+                - 'renyi' for the Rényi entropy.
+                    Rényi entropy formula is the following: H_α = 1/(1-α)*log_2(sum_i(p^α_i)),
+                    where α is a parameter. When α == 1, it works like the Gibbs entropy.
+                    More: https://en.wikipedia.org/wiki/R%C3%A9nyi_entropy
+
+        alpha: Power scale for logsoftmax (α for entropies). Here we restrict it to be > 0.
+            When the alpha equals one, scaling is not applied to 'max_prob',
+            and any entropy type behaves like the Shannon entropy: H = -sum_i(p_i*log(p_i))
+
+        entropy_norm: A mapping of the entropy value to the interval [0,1].
+            Supported values:
+                - 'lin' for using the linear mapping.
+                - 'exp' for using exponential mapping with linear shift.
+    """
+
     name: str = "entropy"
     entropy_type: str = "tsallis"
-    temperature: float = 0.33
+    alpha: float = 0.33
     entropy_norm: str = "exp"
```

Comment:
Revert. You can add alpha, you cannot remove temperature. Dataclass can only add config items not remove them otherwise they break model loading

Reply:
Done